packages/vite/src/node/shortcuts.ts

```diff
@@ -0,0 +1,111 @@
+import colors from 'picocolors'
+import type { ViteDevServer } from './server'
+import { openBrowser } from './server/openBrowser'
+import { isDefined } from './utils'
+
+export type BindShortcutsOptions = {
+  /**
+   * Print a one line hint to the terminal.
+   */
+  print?: boolean
+  customShortcuts?: (CLIShortcut | undefined | null)[]
+}
+
+export type CLIShortcut = {
+  key: string
+  description: string
+  action(server: ViteDevServer): void | Promise<void>
+}
+
+export function bindShortcuts(
+  server: ViteDevServer,
+  opts: BindShortcutsOptions,
+): void {
+  if (!server.httpServer) return
+
+  if (opts.print) {
+    server.config.logger.info(
+      colors.dim(colors.green('  ➜')) +
+        colors.dim('  press ') +
+        colors.bold('h') +
+        colors.dim(' to show help'),
+    )
+  }
+
+  const shortcuts = (opts.customShortcuts ?? [])
+    .filter(isDefined)
+    .concat(BASE_SHORTCUTS)
+
+  let actionRunning = false
+
+  const onInput = async (input: string) => {
+    // ctrl+c or ctrl+d
+    if (input === '\x03' || input === '\x04') {
+      process.emit('SIGTERM')
+      return
+    }
```

Comment:
The [node docs](https://nodejs.org/api/process.html#signal-events) showed that it's commonly emitted as `SIGINT` for all platforms. Seems like `SIGTERM` isn't supported in Windows too. Should we change this to `SIGINT`?

With that said we only listen for `SIGTERM` for the server today (not sure if that has been an issue in Windows), but I think it's nice to follow the default.